### ISSUE TITLE
Create a print view route & page

### ISSUE THREAD

**Issue by johnblackspear** [ProcessMaker/processmaker#2958]
Related to #2956 and dependent on #2957.

See released package https://github.com/ProcessMaker/package-process-documenter/releases/tag/v1.0.1

We need a simple page that displays the BPMN diagram in a way that is printable.

This needs
 - [x] a title
 - [x] the BPMN diagram SVG displayed (via #2957)
 - [x] (future) possible other metadata, like documentation (#2959)

**Comment by velkymx** [comment]
**Mockup**

![image](https://user-images.githubusercontent.com/535300/77098213-3c5b8900-69cf-11ea-9b97-73cb2bffb01d.png)

**Sample Code**

```
<h4>Process Name</h4>
              <p><i class="far fa-clock"></i> Updated 2020-03-20 0700 by <img src=avatar.png> User Name</p>
              <img src="email_catch_process.png" alt="">
              <hr>
              <h4>Process Elements</h4>





              <div class="card mb-2">
                <div class="card-header">
                  <span class="badge badge-light"><i class="fa fa-circle p-1" style="color:green"></i></span> <strong>Element Label</strong> (node_1)
                </div>               
              <div class="card-body">            
                Deadlights jack lad schooner <strong>scallywag</strong> dance the <a href="#">hempen jig carouser</a> broadside cable strike colors. Bring a spring upon her cable holystone blow the man down spanker Shiver me timbers to go on account lookout wherry doubloon chase. Belay yo-ho-ho keelhaul squiffy black spot yardarm spyglass sheet transom heave to.
              </div>
            </div>


              <div class="card">
                <div class="card-header">
                  <span class="badge badge-secondary"><i class="fa fa-file-code p-1"></i></span> 
                  <strong>Check IMAP Server</strong> (node_2)
                </div>                 
                <div class="card-body">
                  <iframe width="560" height="315" src="https://www.youtube.com/embed/odhFiGcqwtw" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
                </div>
              </div>
```

**Comment by PMQABen** [comment]
The listing of process elements seems to be based on what order elements are placed into the modeler. So for example, a start event, a task, and a connector to go between the two will be ordered like so:
Start (node 1)
Task (node 2)
Connector (node 3)

Even if the user edits the node identifiers to make the connector node 2 and the task node 3, it will still retain that layout in the print page.

In other instances, not all the node element descriptions are arranged in numerical order. 
![image](https://user-images.githubusercontent.com/28575916/78178912-37f28f80-7415-11ea-897f-2785dc5756d7.png)
This seems to occur if the user has modified the process flow after reloading the page, so new elements get unused or previously deleted node identifiers. Likely related to the first point.

@velkymx Overall, it works, but the listing part feels like it needs a bit of polish, if possible.

**Comment by johnblackspear** [comment]
Thanks @PMQABen, that's good feedback, we'll discuss and revert.

**Comment by velkymx** [comment]
@PMQABen good point. It should be sorted by node id.

**Comment by johnblackspear** [comment]
Will track sorting via this ticket: https://github.com/ProcessMaker/processmaker/issues/3014

**Comment by swim-mer** [comment]
<img width="1157" alt="Screen Shot 2020-04-08 at 10 25 20 AM" src="https://user-images.githubusercontent.com/28671488/78815118-9aeba580-7984-11ea-8eff-e9777d2d2af5.png">

Failed to load resource: the server responded with a status of 404 ()

**Comment by johnblackspear** [comment]
@swim-mer this has been moved to a separate package by Sanja yesterday, https://github.com/ProcessMaker/package-process-documenter/ - did you test with that package installed?

**Comment by PMQABen** [comment]
Nobody told us this.

**Comment by johnblackspear** [comment]
No worries, thanks for the testing work

**Comment by PMQABen** [comment]
* If none of the nodes have a description, the process image will not display.
* Still not sorting by node ID. I think the changes in https://github.com/ProcessMaker/processmaker/pull/3023/files never got put into the package.

**Comment by velkymx** [comment]
@PMQABen @swim-mer this was merged to a package last week per Brian. I think it's still in the stages of being setup there. The only thing this now adds is an image (SVG) of the process in the DB.

**Comment by PMQABen** [comment]
```
Element '{http://www.omg.org/spec/BPMN/20100524/MODEL}task', attribute 'color': The attribute 'color' is not allowed.
```
BPMN validation warning comes up when using font colors in a node description. This does not prevent the process from being saved, but it will appear every time the process is edited.

With the BPMN auto-validator enabled, deleting a description with color text, and then continuing to rapidly press the delete key, can throw the following console errors:
```
rule <conditional-flows> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <end-event-required> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <event-sub-process-typed-start-event> failed with error:  TypeError: Cannot read property 
rule <label-required> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <no-complex-gateway> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <no-disconnected> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <no-gateway-join-fork> failed with error:  TypeError: Cannot read property 'forEach' of 
rule <no-implicit-split> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <single-event-definition> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <start-event-required> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <sub-process-blank-start-event> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <processmaker/custom-validation> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <processmaker/gateway-direction> failed with error:  TypeError: Cannot read property 'forEach' of undefined
rule <processmaker/call-activity-child-process> failed with error:  TypeError: Cannot read property 'forEach' of undefined
```
Video: [bpmerror.mp4.txt](https://github.com/ProcessMaker/processmaker/files/4477943/bpmerror.mp4.txt)

**Comment by PMQABen** [comment]
@velkymx Are we OK with the validation warning and the console errors I reported?

**Comment by PMQABen** [comment]
The above-mentioned console errors and validation warning needs to be handled.

**Comment by johnblackspear** [comment]
@PMQABen Hi, from the video you posted you were testing inside the modeler/editor view, not in the printable view. 

This seems to be unrelated to the printable view so we've created a separate ticket for it: https://github.com/ProcessMaker/modeler/issues/1211

If that misses any issues you picked while testing modeler please shout and we'll add them.

And, of course, same if the print view has issues :)
